### PR TITLE
fix: a human-readable error when lock cannot be acquired

### LINKED ISSUE #8709: Why does the compilation fail during installation
**Context:**
- Playwright Version: latest
- Operating System: big sur
- Node.js version: 16.7.0
- Browser: none
- Extra: none

image

As shown, I'm tried to install playwright. But this throws an error. Why?

### PR DESCRIPTION
**Pull request by aslushnikov**
image

Fixes #8709

### CODE REVIEW

File: src/utils/registry.ts [hunk 1]
```diff
@@ -562,6 +562,25 @@ export class Registry {
...
+          `- wait a few minutes if other Playwright is installing browsers in parallel`,
+          `- remove lock manually with:`,
+          ``,
+          `    ${rmCommand} ${lockfilePath}`,
+          ``,
+          `<3 Playwright Team`,
+        ].join('\n'), 1));
+      } else {
+        throw e;
+      }
```
Comment (mxschmitt) [inline src/utils/registry.ts:583]:
```suggestion
      }
      throw e;
```
nit

File: src/utils/registry.ts [hunk 2]
```diff
@@ -562,6 +562,25 @@ export class Registry {
         else
           throw new Error(`ERROR: Playwright does not support installing ${executable.name}`);
       }
+    } catch (e) {
+      if (e.code === 'ELOCKED') {
+        const rmCommand = process.platform === 'win32' ? 'rm -R' : 'rm -rf';
```
Comment (mxschmitt) [inline src/utils/registry.ts:567]: nit: do we need `-r` there?

**Review by mxschmitt — APPROVED** [review]

File: src/utils/registry.ts (hunk 1 above)
Comment (aslushnikov) [reply src/utils/registry.ts:583]: has to be in parenthesis since the first `if` clause is multi-line.

Comment (aslushnikov) [reply src/utils/registry.ts:583]: ah, i'd rather have it this way i guess!

File: src/utils/registry.ts (hunk 2 above)
Comment (aslushnikov) [reply src/utils/registry.ts:567]: just in case we have a lockfile actually inside the lock directory at some future point of time.

**Pull request merged**